### PR TITLE
[PS-2827] Add Leaderboards

### PR DESCRIPTION
**Pull request by ellsclytn**
Yaaay. Placeholder text on the leaderboards that are invalid (until November).
screen shot 2017-08-31 at 2 26 37 pm
screen shot 2017-08-31 at 2 26 47 pm

### CODE REVIEW

File: source/components/CompactLeaderboard/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+type SuppliedProps = {
+  classNames: any,
+  styles: any
+}
+
+const CompactLeaderboard = ({
+  leaders,
+  styles
+}: SuppliedProps & Props) => (
+  <div>
```
Comment (kizza) [inline source/components/CompactLeaderboard/index.js]: Do we need to wrap in a `div` here?

File: source/components/Leaderboards/index.js [hunk 2]
```diff
@@ -1,12 +1,86 @@
...
+            <p className={classNames.comingSoon}>Cycling positions coming November</p>
+          </div>
+        </div>
+      )
+    )}
+  </section>
 )
 
-export default Leaderboards
+export default (compose(
```
Comment (kizza) [inline source/components/Leaderboards/index.js]: 💯

File: source/components/Leaderboards/index.js [hunk 3]
```diff
@@ -1,12 +1,86 @@
...
+        <button
+          onClick={onToggleOn}
+          className={`${classNames.button} ${toggled ? classNames.activeButton : ''}`}
+        >
+          Individuals
+        </button>
+      </div>
+    </div>
+    {!toggled ? (
+      teams && teams.funds && (
```
Comment (kizza) [inline source/components/Leaderboards/index.js]: What do you think about abstracting these into seperate components (_even_ within the same file)?

Comment (kizza) [reply source/components/Leaderboards/index.js]: Whenever I see render functions this long (especially within conditionals), I can't help but feel that they're more easily grokked when seperated into seperate components (ie. they're all functions anyway).  Thoughts?

File: source/routes/Home/index.js [hunk 4]
```diff
@@ -124,6 +125,7 @@ const Root = ({
             competition={competition}
             endMessage={config.endmessage}
           />}
+          <Leaderboards {...leaderboards} {...leaderboardsContent} />
```
Comment (kizza) [inline source/routes/Home/index.js]: ...this does feel weirder here (to me), I get that `leaderboards` is a defined constant, but destructuring `leaderboards` as well as `leaderboardsContent` is funny (given all other components just destructure one variable?).  Thoughts?

Comment (ellsclytn) [reply source/routes/Home/index.js]: For this one, it boils down to the fact there are two data sources for this component: The CMS data, and Supporter data. _Maybe_ we'd be better off not pulling `leaderboards` (`leaderboardsContent`) out of `home`, and simply spreading `...home.leaderboards` instead, for readability's sake?

File: source/components/Leaderboards/index.js (hunk 3 above)
Comment (ellsclytn) [reply source/components/Leaderboards/index.js]: Good idea.

**Pull request merged**